clients/src/main/java/org/apache/kafka/common/MetricName.java

```diff
@@ -75,53 +74,6 @@ public MetricName(String name, String group, String description, Map<String, Str
         this.tags = Utils.notNull(tags);
     }
 
-    /**
-     * @param name          The name of the metric
-     * @param group         logical group name of the metrics to which this metric belongs
-     * @param description   A human-readable description to include in the metric
-     * @param keyValue      additional key/value attributes of the metric (must come in pairs)
-     */
-    public MetricName(String name, String group, String description, String... keyValue) {
```

Comment:
@junrao I see. Yeah I shouldn't remove these constructors. I have added them back and tagged them as deprecated in the updated patch. The latest patch passed all unit tests and style check. I am redoing the test with multiple producers/consumers to validate that jmx metric names are the same.

Initially I wasn't strong that we should deprecate the MetricName constructors. But now that we construct MetricName only via Metrics, I think it is reasonable to tag these constructors as deprecated so that moving forward MetricName are constructed in the same pattern. What do you think?